Sprint_4

Marval-1 opened the pull request.

Marval-1 committed:
Sprint_4

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,46 @@
+
+@pytest.fixture
+def driver():
+    driver = webdriver.Firefox(executable_path='./geckodriver') # На моей машине работает, если так прописан путь
```
Он будет работать работать на любой машине, если драйвер лежит в корне проекта. Чтобы не городить это необходимо прописать директорию с драйвером в path в системные переменные. 
Но главное тут: никогда не использовать фразы: "На моей машине работает". Проекты пишутся для глобального использования. Как это все работает на локальной машине разработчика никого не интересует. Все должно одинаково подниматься и работать если склонить репозиторий на любую машину и установить зависимости

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,26 @@
+
+    def scroll_down(self, locator):
+        self.driver.execute_script("arguments[0].scrollIntoView();", locator)
+        time.sleep(1)
```
Нужно исправить: использование слипов и неявных ожиданий - плохая практика, это сильно увеличивает время прогона тестов. С одного-двух ты разницы не увидишь, с 1000 - получишь от получаса до 2 часов плюсом.
Это никого не устроит. Присмотрись к использованию expected_conditions

AlexKitIt commented on pages/scooter_main_page.py:
```diff
@@ -0,0 +1,46 @@
+        cookie_button = self.find_element(LocatorScooterMainPage.cookie_button, time=3)
+        cookie_button.click()
+
+    def click_question_element(self, locator):
```
Нужно исправить: а где аннотации allure step над остальными методами?
Исправления из коммента вносить далее по коду

AlexKitIt commented on test_data.py:
```diff
@@ -0,0 +1,27 @@
+class MainPageQuestionsData:
```
Отлично: хорошее решение о выносе data в отдельный модуль

AlexKitIt commented on tests/test_order_flow.py:
```diff
@@ -0,0 +1,61 @@
+        scooter_order_rent_page.filling_out_the_rent_form()
+        text = scooter_order_rent_page.confirm_order()
+        scooter_order_rent_page.click_button_order_status()
+        assert 'Заказ оформлен' in text
```
Можно лучше: если добавишь assert message, то при разборе будет проще и понятнее почему, а главное что-именно у нас упало

AlexKitIt commented on tests/test_order_flow.py:
```diff
@@ -0,0 +1,61 @@
+
+        scooter_main_page = ScooterMainPage(driver)
+        title = scooter_main_page.get_main_title()
+        assert 'Самокат' in title
+
+        scooter_order_status_page.click_logo_yandex(driver)
+        scooter_order_status_page.switch_to_another_tab(driver)
+        url = Urls()
+        assert driver.current_url == url.yandex_url
```
Можно лучше: тут не было смысла явно создавать объект класса, чтобы просто обратиться к его атрибуту. Urlы().yandex_url отработает точно так же. А если еще вытащишь его из init наверх, то тогда даже так создавать объект будет не надо т.к. будем обращаться к статичной переменной

AlexKitIt replied:
Можно лучше: 2 assert в одном тесте не очень хорошо. Если завалится на первом, как пройдет вторая проверка мы уже не узнаем, потому что она не выполнится

AlexKitIt commented on tests/test_order_flow.py:
```diff
@@ -0,0 +1,61 @@
+
+@allure.story('TestOrderFlow')
+class TestOrderFlow:
+    @allure.feature('test order by using button from the header')
```
Можно лучше: тут уместнее аннотация title, feature  - это больше про тестовый класс

AlexKitIt commented on tests/test_questions_main_page.py:
```diff
@@ -0,0 +1,87 @@
+
+
+@allure.story('TestQuestions')
+class TestQuestions:
```
Нудно исправить: этот кейс надо бы сделать один и параметризованный. На вход: локатор вопроса, локатор ответа и текст для сравнения. Только нужно будет локаторы немного переписать, чтобы в них не было явной привязки к тексту вопроса\ответа. Или сваять что-то с format строки

Marval-1 replied:
А в условии проекта указано: "Важно написать отдельный тест на каждый вопрос"
Всё равно сделать один параметризованный или оставить так?

AlexKitIt replied:
Так он и будет отдельным тестом. Тестовый метод будет один, а тестов 8

Marval-1 replied:
Точно.

Marval-1 committed:
Edits after codereview
- Added Allure steps
- Removed implicit expectations
- Made a parametrized question test
- Commit allure results
- Other edits
